cmake/toolchain/xcc/generic.cmake

```diff
@@ -6,6 +6,7 @@ set(COMPILER xcc)
 set(OPTIMIZE_FOR_DEBUG_FLAG "-O0")
 set(CC xcc)
 set(C++ xc++)
+set(LINKER xt-ld)
```

Comment:
Do we really need to use an obsolete linker in addition to using an obsolete compiler? The existing solution worked fine until the big CMake shuffle. Moreover, you seemed to have cracked the TOOLCHAIN_LIBS nut in v2 so is this switch really still necessary?

Reply:
I think it is better to stay within the confine of the Xtensa toolchain instead of mixing toolchain components that may not be compatible with each other.

Reply:
I think this is debatable. Compilation and linkage are two steps fairly independent from each other with good compatibility and zephyr/cmake seems to assume they are indeed. `xt-xcc` has been a pre-C11 pain but it features proprietary customizations found no one else, so it's a _necessary_ pain. What value does `xt-ld` bring to the table besides old linker bugs that no-one wants to debug and a lack of recent features? Isn't the same than `ld`, just older? Also: it has apparently NOT been used in production. Probably by accident but still.

But back to the initial issue: can't we have ONLY a fix for the CMake regression FIRST? And then only later a linker change once SOF CI is back on its feet?